Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -318,6 +318,12 @@ public PointerEventData GraphicEventData
             }
             private PointerEventData graphicData;
 
+            /// <summary>
+            /// Returns true if the current pointer target has been disabled or destroyed
+            /// </summary>
+            public bool IsCurrentPointerTargetInvalid => ((CurrentPointerTarget != null && !CurrentPointerTarget.activeInHierarchy)) ||
+                (CurrentPointerTarget == null && !ReferenceEquals(CurrentPointerTarget, null));
```

Comment:
Why is reference equals here?

Reply:
`GameObjects` have overloaded equality, so that they return true for `gameObject == null` once the object is destroyed. However, the object hasn't been cleaned up yet, so destroyed objects will not reference equal null:
http://answers.unity.com/answers/1001265/view.html

Reply:
If we only checked `gameObject == null` then this would also be true if the pointer no longer points at the focus object. If we're focus locking, no longer pointing at the focus object shouldn't end focus lock